specification/network/resource-manager/Microsoft.Network/stable/2020-03-01/ipAllocation.json

```diff
@@ -0,0 +1,395 @@
+        },
+        "x-ms-examples": {
+          "List all IpAllocations": {
+            "$ref": "./examples/IpAllocationList.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/IpAllocations": {
+      "get": {
+        "tags": [
+          "IpAllocations"
+        ],
+        "operationId": "IpAllocations_ListByResourceGroup",
+        "description": "Gets all IpAllocations in a resource group.",
+        "parameters": [
+          {
+            "name": "resourceGroupName",
+            "in": "path",
+            "required": true,
+            "type": "string",
+            "description": "The name of the resource group."
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Request successful. The operation returns a list of IpAllocation resources.",
+            "schema": {
+              "$ref": "#/definitions/IpAllocationListResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "./network.json#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "List IpAllocations in resource group": {
+            "$ref": "./examples/IpAllocationListByResourceGroup.json"
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "IpAllocationPropertiesFormat": {
+      "properties": {
+        "subnet": {
+          "$ref": "./virtualNetwork.json#/definitions/Subnet",
+          "description": "The Subnet that using the prefix of this IpAllocation resource."
+        },
+        "virtualNetwork": {
+          "$ref": "./virtualNetwork.json#/definitions/VirtualNetwork",
+          "description": "The VirtualNetwork that using the prefix of this IpAllocation resource."
+        },
+        "type": {
+          "$ref": "#/definitions/IpAllocationType",
+          "description": "The type for the IpAllocation."
+        },
+        "prefix": {
+          "type": "string",
+          "description": "The address prefix for the IpAllocation."
+        },
+        "prefixLength": {
+          "type": "integer",
+          "x-nullable": true,
+          "default": 0,
+          "description": "The address prefix length for the IpAllocation."
+        },
+        "prefixType": {
+          "$ref": "./network.json#/definitions/IPVersion",
+          "default": null,
+          "description": "The address prefix Type for the IpAllocation."
+        },
+        "ipamAllocationId": {
```

Comment:
>"ipamAllocationId" [](start = 8, length = 18)

where does the user get this?

Reply:
This is a IPAM service side ID, in some scenario, IPAM service will create data for user first, and provide the ipamAllocationId to user and user then create IpAllocation resource by the ipamAllocationId.